### PR TITLE
Fix jobs that have a retryAfter and a repeat 'scheduled', take 2

### PR DESCRIPTION
**Pull request by flodnv**
Followup to https://github.com/Expensify/Bedrock/pull/912

As discussed in https://github.com/Expensify/Expensify/issues/143711#issuecomment-742508080

Updated the automated test.

### CODE REVIEW

File: test/tests/jobs/GetJobsTest.cpp [hunk 1]
```diff
@@ -108,8 +108,9 @@ struct GetJobsTest : tpunit::TestFixture {
 
             // Let's see if it's scheduled at the right time.
             if (stoull(row[0]) == jobIDs[0]) {
-                // Assert that the difference between "lastRun + 1hour" and "nextRun" is less than 3 seconds.
-                ASSERT_LESS_THAN(absoluteDiff(stringToUnixTimestamp(row[2]) + 1 * 60 * 60, stringToUnixTimestamp(row[3])), 3);
+                // Assert that the difference between "lastRun + 1hour - 5 seconds" and "nextRun" is less than 3 seconds.
+                // We remove 5 seconds here because the job is rescheduled for when it was _scheduled_, not when it started/finished
```
Comment (flodnv) [inline test/tests/jobs/GetJobsTest.cpp:112]: Please double check me on this.... After some time debugging why `nextRun` was always 5 seconds before `lastRun + 1hour`, I actually realized that my change **actually** fixes the broken behavior: we now reschedule the job with regards to its SCHEDULED time, not the time at which it finishes (see the sleep(5) on line 91)

File: plugins/Jobs.cpp [hunk 2]
```diff
@@ -920,7 +922,7 @@ void BedrockJobsCommand::process(SQLite& db) {
 
         // Verify there is a job like this and it's running
         SQResult result;
-        if (!db.read("SELECT state, nextRun, lastRun, repeat, parentJobID, json_extract(data, '$.mockRequest'), retryAfter "
+        if (!db.read("SELECT state, nextRun, lastRun, repeat, parentJobID, json_extract(data, '$.mockRequest'), retryAfter, json_extract(data, '$.originalNextRun') "
```
Comment (iwiznia) [inline plugins/Jobs.cpp:925]: Do we need to clear the originalNextRun at some point or it is not necessary?
I _think_ we don't because:
- If the job did not have a repeat, it will be deleted when finished
- If it has a repeat, it will get re-scheduled and the originalNextRun re-set when retrieved in getjob

**Review by iwiznia — DISMISSED** [review]

**Comment by tylerkaraszewski** [comment]
This seems like the opposite to the fix I made here: https://github.com/Expensify/Bedrock/pull/904

**Review by tylerkaraszewski — APPROVED** [review]
This code is always a PITA to work with, but I *think* this works OK.

File: plugins/Jobs.cpp (hunk 2 above)
Comment (flodnv) [reply plugins/Jobs.cpp:925]: Agreed, thought the same

**Comment by flodnv** [comment]
>This seems like the opposite to the fix I made here: #904

Hmmm, is it? That PR fixed `nextRun` inside `GetJob` (so right before the job actually runs), this PR is fixing `nextRun` inside `FinishJob` (so right after the job ran) in a corner case scenario. 

Updated the PR to fix a related failing test.

**Review by iwiznia — APPROVED** [review]

**Review by tylerkaraszewski — APPROVED** [review]
